### PR TITLE
Update Botocore to 1.34.131

### PR DESCRIPTION
**Pull request by lou-k**
### Description of Change
Updates botocore to 1.34.131

There is no added functionality in 1.34.131:
```
git diff --name-status 1.34.106...1.34.131 botocore/ | grep py$
M	botocore/__init__.py
M	botocore/retries/standard.py
M	botocore/utils.py
```

but I'd like to use the latest `boto3` in my project, and the existing `<1.34.107` is causing a dependency issue.


### Assumptions
*I got some errors on `make cov` before I made any changes. I'm assuming that is an issue with my local environment and CICD will pass here...*

### Checklist for All Submissions
* [x] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)
* ~[ ] If this is resolving an issue (needed so future developers can determine if change is still necessary and under what conditions) (can be provided via link to issue with these details):~
* ~[ ] If this is providing a new feature  (can be provided via link to issue with these details):~

### Checklist when updating botocore and/or aiohttp versions

* [x] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [x] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [x] I have ensured that the awscli/boto3 versions match the updated botocore version


### Tests Run Locally

```
pytest -sv tests/test_patches.py 
 ======================================== test session starts  ========================================
platform darwin -- Python 3.12.3, pytest-8.1.1, pluggy-1.5.0 -- /Users/lou/.local/share/virtualenvs/aiobotocore-UGsiy0fU/bin/python
cachedir: /tmp/pytest_aiobotocore_cache
rootdir: /Users/lou/repos/aiobotocore
configfile: pyproject.toml
plugins: cov-5.0.0, asyncio-0.21.2, xdist-3.5.0
asyncio: mode=Mode.AUTO
collected 2 items                                                                                                                                                                                                                                            

tests/test_patches.py::test_protocol_parsers PASSED
tests/test_patches.py::test_patches Botocore version: 1.34.131
PASSED

 ======================================== 2 passed in 0.20s ========================================

```

Thanks for the easy-to-follow guide on updating this!

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,15 +7,15 @@
 # NOTE: When updating botocore make sure to update awscli/boto3 versions below
 install_requires = [
     # pegged to also match items in `extras_require`
-    'botocore>=1.34.70,<1.34.107',
```
Comment (jakob-keller) [inline setup.py:10]: Since there appear to be no relevant upstream changes, then lower bound can be kept at 1.34.70.

File: setup.py [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
+    'botocore>=1.34.107,<1.34.132',
     'aiohttp>=3.9.2,<4.0.0',
     'wrapt>=1.10.10, <2.0.0',
     'aioitertools>=0.5.1,<1.0.0',
 ]
 
 extras_require = {
-    'awscli': ['awscli>=1.32.70,<1.32.107'],
```
Comment (jakob-keller) [inline setup.py:17]: Since there appear to be no relevant upstream changes, then lower bound can be kept at 1.32.70 and 1.34.70 respectively.

**Review by jakob-keller — CHANGES_REQUESTED** [review]

**Comment by lou-k** [comment]
thanks for the eyes, @jakob-keller !

**Review by jakob-keller — APPROVED** [review]

**Comment by jakob-keller** [comment]
closes #1123

**Comment by thehesiod** [comment]
oh we hit this error at work, need to upgrade docker/requests IIRC

**Comment by thehesiod** [comment]
ughh, ci tool

**Comment by thehesiod** [comment]
sweet!

**Pull request merged**